fixture-monkey-jakarta-validation/src/main/java/com/navercorp/fixturemonkey/jakarta/validation/introspector/JakartaValidationConstraintGenerator.java

```diff
@@ -292,178 +292,126 @@ public JavaIntegerConstraint generateIntegerConstraint(ArbitraryGeneratorContext
-				negativeMax = BigDecimal.ZERO;
-				negativeMaxInclusive = false;
+			if (max == null || BigDecimal.ZERO.compareTo(max) < 0
+				|| (BigDecimal.ZERO.compareTo(max) == 0 && maxInclusive)) {
+				max = BigDecimal.ZERO;
+				maxInclusive = false;
 			}
 		}
 
 		if (context.findAnnotation(NegativeOrZero.class).isPresent()) {
-			if (negativeMax == null || negativeMax.compareTo(BigDecimal.ZERO) > 0) {
-				negativeMax = BigDecimal.ZERO;
-				negativeMaxInclusive = true;
+			if (max == null || BigDecimal.ZERO.compareTo(max) < 0) {
+				max = BigDecimal.ZERO;
+				maxInclusive = true;
 			}
 		}
 
-		if (context.findAnnotation(Positive.class).isPresent()) {
-			if (positiveMin == null || positiveMin.compareTo(BigDecimal.ZERO) < 0) {
-				positiveMin = BigDecimal.ZERO;
-				positiveMinInclusive = false;
+		Optional<Digits> digitsAnn = context.findAnnotation(Digits.class);
+		if (digitsAnn.isPresent()) {
+			Digits digits = digitsAnn.get();
+			int integerDigits = digits.integer();
+			int fractionDigits = digits.fraction();
+
+			StringBuilder maxBuilder = new StringBuilder();
+			for (int i = 0; i < integerDigits; i++) {
+				maxBuilder.append('9');
 			}
-		}
+			if (fractionDigits > 0) {
+				maxBuilder.append('.');
+				for (int i = 0; i < fractionDigits; i++) {
+					maxBuilder.append('9');
+				}
+			}
+			BigDecimal digitsMax = new BigDecimal(maxBuilder.toString());
+			BigDecimal digitsMin = digitsMax.negate();
 
-		if (context.findAnnotation(PositiveOrZero.class).isPresent()) {
-			if (positiveMin == null || positiveMin.compareTo(BigDecimal.ZERO) < 0) {
-				positiveMin = BigDecimal.ZERO;
-				positiveMinInclusive = true;
+			if (max == null || digitsMax.compareTo(max) < 0) {
+				max = digitsMax;
+				maxInclusive = true;
+			}
+			if (min == null || digitsMin.compareTo(min) > 0) {
+				min = digitsMin;
+				minInclusive = true;
 			}
+
+			scale = digits.fraction();
 		}
 
-		if (positiveMin == null && positiveMax == null && negativeMin == null && negativeMax == null && scale == null) {
+		if (min == null && max == null) {
 			return null;
 		}
 
+		boolean isPositiveMin = min != null && min.compareTo(BigDecimal.ZERO) >= 0;
+		boolean isPositiveMax = max != null && max.compareTo(BigDecimal.ZERO) >= 0;
+		boolean isNegativeMin = min != null && min.compareTo(BigDecimal.ZERO) < 0;
+		boolean isNegativeMax = max != null && max.compareTo(BigDecimal.ZERO) < 0;
+
 		return new JavaDecimalConstraint(
-			positiveMin,
-			positiveMinInclusive,
-			positiveMax,
-			positiveMaxInclusive,
-			negativeMin,
-			negativeMinInclusive,
-			negativeMax,
-			negativeMaxInclusive,
+			isPositiveMin ? min : null,
+			isPositiveMin ? minInclusive : null,
+			isPositiveMax ? max : null,
+			isPositiveMax ? maxInclusive : null,
+
+			isNegativeMin ? min : null,
+			isNegativeMin ? minInclusive : null,
+			isNegativeMax ? max : null,
+			isNegativeMax ? maxInclusive : null,
 			scale
 		);
```

Comment:
I don't understand why the JavaDecimalConstraint class needs to manage separate min/max values for negative and positive numbers. In the previous implementation with the @Digits annotation, this was necessary, but with the "3. Improved @Digits Annotation Processing," the method for enforcing maximum digit limits has changed. In this case, wouldn't it be sufficient to just manage the min/max values and their inclusiveness to generate random numbers? What do you think about changing this approach?

Reply:
Is there a better idea for addressing both positive and negative values in the case of `@Digits`?

For example, in the case of `@Digits(integer=3, fraction=2)`, it should be the value that satisfies the following constraints.

1. 0 <= value <= 999.99
2. -999.99 <= value <= 0

Reply:
Looking at your example with `@Digits(integer=3, fraction=2)`, I believe we could satisfy these constraints simply using a minimum (-999.99) and maximum (999.99) value.
Is there any specific case where we need to manage positive and negative ranges separately in the current implementation? If I'm missing something important, please let me know.